src/dbms/dbms_handler.cpp

```diff
@@ -9,58 +9,214 @@
 // by the Apache License, Version 2.0, included in the file
 // licenses/APL.txt.
 
+#ifdef MG_ENTERPRISE
+
 #include "dbms/dbms_handler.hpp"
 
+#include <cstdint>
+#include <filesystem>
+
+#include "dbms/constants.hpp"
+#include "dbms/replication_client.hpp"
+#include "spdlog/spdlog.h"
+#include "utils/exceptions.hpp"
+#include "utils/uuid.hpp"
+
+namespace {
+constexpr std::string_view kDBPrefix = "database:";  // Key prefix for database durability
+}  // namespace
 namespace memgraph::dbms {
-#ifdef MG_ENTERPRISE
+
+struct Durability {
+  enum class DurabilityVersion : uint8_t {
+    V0 = 0,
+    V1,
+  };
+
+  struct VersionException : public utils::BasicException {
+    VersionException() : utils::BasicException("Unsupported durability version!") {}
+  };
+
+  struct UnknownVersionException : public utils::BasicException {
+    UnknownVersionException() : utils::BasicException("Unable to parse the durability version!") {}
+  };
+
+  struct MigrationException : public utils::BasicException {
+    MigrationException() : utils::BasicException("Failed to migrate to the current durability version!") {}
+  };
+
+  static DurabilityVersion StringToVersion(std::string_view str) {
+    if (str == "V0") return DurabilityVersion::V0;
+    if (str == "V1") return DurabilityVersion::V1;
+    throw VersionException();
+  };
+
+  static DurabilityVersion VersionCheck(std::optional<std::string_view> val) {
+    if (!val) {
+      return DurabilityVersion::V0;
+    }
+    if (val == "V1") {
+      return DurabilityVersion::V1;
+    }
+    throw UnknownVersionException();
+  };
+
+  static auto GenKey(std::string_view name) -> std::string { return fmt::format("{}{}", kDBPrefix, name); }
+
+  static auto GenVal(utils::UUID uuid, std::filesystem::path rel_dir) {
+    nlohmann::json json;
+    json["uuid"] = uuid;
+    json["rel_dir"] = rel_dir;
+    // TODO: Serialize the configuration
```

Comment:
Finish the TODO

Reply:
Not sure how to handle differences between the saved config and incoming command line flags.
This is a new feature and we need to decide how to handle it.

Reply:
Decide to leave as is. Problem we will come back to in the future.